dcc/utils.py

```diff
@@ -0,0 +1,80 @@
+"""``dcc`` utils."""
+import json
+import os
+import subprocess
+import sys
+
+from . import BASE_DIR, HUB, PIP, VIRTUALENV, DjangoCCError
+
+PY3 = sys.version_info[0] == 3
+if PY3:
+    from urllib.request import urlopen
+else:
+    from urllib import urlopen
+
+
+PYPI_JSON_URL = "https://pypi.python.org/pypi/{package_name}/json"
+GITHUB_URL = "https://github.com/{username}/{repo_name}.git"
+
+
+def get_github_url(package_name):
+    """Return Github url based on PYPI package information."""
+    url = PYPI_JSON_URL.format(package_name=package_name)
+    response = urlopen(url)
+    if response.code == 404:
+        raise DjangoCCError('The package "%s" does not exist.' % package_name)
+    if PY3:
+        body = response.read().decode('utf-8')
+    else:
+        body = response.read()
+    data = json.loads(body)
+    github_url = data['info']['download_url']
+    if not github_url.startswith('https://github.com/'):
+        raise DjangoCCError('The package "%s" does not support Django-CC.' % package_name)
+    return github_url
+
+
+def parse_github_url(url):
+    """Return username and repository name base on Github url."""
+    return [x for x in url.rsplit('/') if x][-2:]
+
+
+def setup_git(package_name, github_url):
+    """Clone git and setup own Github fork and remote."""
+    owner_name, repo_name = parse_github_url(github_url)
+    github_url = GITHUB_URL.format(username=owner_name, repo_name=repo_name)
+    print("Cloning %s" % github_url)
+    subprocess.check_call([HUB, 'clone', github_url, repo_name, '-o', owner_name, '-q'])
+    try:
+        print("Forking...")
+        subprocess.check_call([HUB, 'fork'], cwd=repo_name)
+    except subprocess.CalledProcessError:
+        print("Fork already exists. Adding remote...")
+        output = subprocess.check_output([HUB, 'config', '--get', 'github.user'])
+        username = output.strip().decode('utf-8')
+        if username != owner_name:
+            fork_url = GITHUB_URL.format(username=username, repo_name=repo_name)
+            subprocess.check_call([HUB, 'remote', 'add', username, fork_url], cwd=repo_name)
```

Comment:
That is a very crucial question. I do prefer SSH especially since HTTPS and oauth doesn't really work well with two factor authentication. Do you know if SSH works out of the box with GitHub Desktop? It's meant to be a tool for beginners.